horaedb/metric_engine/src/storage.rs

```diff
@@ -69,6 +79,7 @@ pub struct CloudObjectStorage {
     path: String,
     store: ObjectStoreRef,
     arrow_schema: SchemaRef,
+    primary_key_indexs: Vec<usize>,
```

Comment:
I think we use `num_primary_key: usize`, it's more easier to use, and have no heap memory allocation.
`0..num_primary_keys` columns are primary key.